src/main/webapp/app/quiz/re-evaluate/quiz-re-evaluate-warning.controller.js

```diff
@@ -0,0 +1,132 @@
+                var backUpQuestion = null;
+                for (var i = 0; backUpQuestion === null && i < backUpQuiz.questions.length; i++) {
+                    if (backUpQuiz.questions[i].id === question.id) {
+                        backUpQuestion = backUpQuiz.questions[i];
+                    }
+                }
+                if (backUpQuestion != null) {
+                    // question set invalid?
+                    if (question.invalid !== backUpQuestion.invalid) {
+                        vm.questionInvalid = true;
+                    }
+                    // answer deleted?
+                    if (question.answerOptions.length !== backUpQuestion.answerOptions.length) {
+                        vm.answerDeleted = true;
+                    }
+                    // question scoring changed?
+                    if (question.scoringType !== backUpQuestion.scoringType) {
+                        vm.scoringChanged = true;
+                    }
+                    //check each answer
+                    question.answerOptions.forEach(function (answer) {
+                        // only check if there are no changes on the answers yet
+                        if (!vm.answerCorrectess || !vm.answerInvalid) {
+                            var backUpAnswer = null;
+                            for (var j = 0; backUpAnswer === null && j < backUpQuestion.answerOptions.length; j++) {
+                                if (backUpQuestion.answerOptions[j].id === answer.id) {
+                                    backUpAnswer = backUpQuestion.answerOptions[j];
+                                }
+                            }
+                            if (backUpAnswer != null) {
+                                //answer set invalid?
+                                if (answer.invalid !== backUpAnswer.invalid) {
+                                    vm.answerInvalid = true;
+                                }
+                                //answer correctness changed?
+                                if (answer.isCorrect !== backUpAnswer.isCorrect) {
+                                    vm.answerCorrectess = true;
+                                }
+                            }
+                        }
+                    });
+                }
+            });
+        }
+
+        /**
+         * close modal
+         */
+        function clear() {
+            $uibModalInstance.dismiss('cancel');
+        }
+
+        /**
+         * Confirm changes
+         *  => send changes to server and wait for result
+         *  if saving failed -> show failed massage
+         */
+        function confirmChange() {
+
+            QuizExerciseReEvaluate.update(vm.quizExercise,
+                function () {
+                    vm.successful = true;
+                },
+                function () {
+                    vm.failed = true;
+                });
+        }
```

Comment:
please add an indicator here, that the server is working, e.g. https://github.com/cgross/angular-busy